lib/components/auth/digest.ts

```diff
@@ -73,16 +73,19 @@ export class DigestAuth {
   ha1 = (cnonce: string): string => {
     let ha1 = this.ha1Base
     if (this.algorithm === 'md5-sess') {
-      ha1 = new MD5().update(`${ha1}:${this.nonce}:${cnonce}`).digest('hex')
+      ha1 = new MD5()
```

Comment:
```javascript
return this.algorithm === 'md5-sess' ? new MD5().update(...) : this.ha1Base
```

I find it _slightly_ confusing that you have variables with the same name as functions. Both here and in other places.

Reply:
I realize it's not great, but I would like to keep the structure in place when just replacing the dependency, otherwise it's even more difficult to trace the changes.

Reply:
Fair enough!